src/app/util/util-test.ts

```diff
@@ -0,0 +1,24 @@
+import { ComponentFixture } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { NavigationEnd, Router } from '@angular/router';
+
+import { BehaviorSubject } from 'rxjs';
+
+const mockRouter = () => {
+  const routerNavigationEndSubject = new BehaviorSubject(new NavigationEnd(1, 'test', 'test'));
+
+  return {
+    events: routerNavigationEndSubject.asObservable(),
+    navigate: jasmine.createSpy('navigate'),
+  }
```

Comment:
@Yogano To be honest, at the beginning I wanted just to use RoutingTestingModule without implementing of custom mock router provider. However, I faced with some warnings when tried to navigate from within one of test suites. Therefore I decided to implement custom router provider. In order to pass various paths I think that we can reimplement the object returned by `mockRouter` function in following way:

Instead of this
```
  return {
    events: routerNavigationEndSubject.asObservable(),
    navigate: jasmine.createSpy('navigate'),
  }
```
we can write something like that
```
  return {
    events: (() => routerNavigationEndSubject.asObservable())(),
    navigate: ([ path ]) => {
        const formattedPath = path.replace(path, '/')
        routerNavigationEndSubject.next(new NavigationEnd(1, formattedPath, formattedPath))
    }
  }
```
I assume that reference to routerNavigationEndSubject should be in closure for both anonymous functions-handlers.

If example above doesn't work for some reason, we can try also another class-based approach:
```
class MockRouter {
  private routerNavigationEndSubject = new BehaviorSubject(new NavigationEnd(1, 'test', 'test'));

  get events() {
    return routerNavigationEndSubject.asObservable()
  },
  
  navigate([ path ]) {
    const formattedPath = path.replace(path, '/')
    this.routerNavigationEndSubject.next(new NavigationEnd(1, formattedPath, formattedPath))
  },
}
```
and then export our provider in following way:
```
export const mockRouterProvider = { provide: Router, useClass: MockRouter }
```

I didn't try out mentioned options but I believe that at least one of them should work.